### PR TITLE
GetNoNIntersectedSpans refactoring

### PR DESCRIPTION
**Pull request by jonyrock**
We have method `GetNoNIntersectedSpans` which takes a strange array of "borders".
The goal of this PR is to make `GetNoNIntersectedSpans` function more robust and clear

## Changes
* `Getnonintersectedspans`  -> `cutSpanWithSpans`
* `cutSpanWithSpans` takes and returns `Spans`, not numbers
* `spans.jest.ts` refactoring according to new signature of `cutSpanWithSpans`
* `cutSpanWithSpans` can take ANY set of spans: not requirement of being sorted or not-overlapping
* remove method `getSpanBorders` (and tests) for it

### CODE REVIEW

File: server/src/utils/spans.ts [hunk 1]
```diff
@@ -2,53 +2,66 @@
...
-        result.push({ from: startDetectionRange, to: border});
-      }
+  console.log('mergedSortedCuts')
+  console.log(mergedSortedCuts)
+
+
+  // this is what we get if we cut `mergedSortedCuts` from (-Infinity, Infinity)
+  var holes = mergedSortedCuts.map((cut, i) => {
+    var from = -Infinity;
+    var to = cutSpans[0].from;
```
Comment (rozetko) [inline server/src/utils/spans.ts]: let's use `let`s instead of `var`s here

File: server/src/controllers/analytics_controller.ts [hunk 2]
```diff
@@ -564,9 +564,7 @@ export async function getDetectionSpans(
     }
   }
 
-  const spanBorders = Detection.getSpanBorders(readySpans);
-
-  let newDetectionSpans = getNonIntersectedSpans(from, to, spanBorders);
+  let newDetectionSpans = cutSpanWithSpans({ from, to }, readySpans);
```
Comment (rozetko) [inline server/src/controllers/analytics_controller.ts]: `cutSpanWithSpans` takes `Span` as a 1st argument but `{ from, to }` is not a `Span`

File: server/spec/utils/spans.jest.ts [hunk 3]
```diff
@@ -1,35 +1,47 @@
...
-    expect(getNonIntersectedSpans(4, 20, spanBorders)).toEqual([{from: 5, to: 6}, {from: 8, to: 10}]);
-    expect(getNonIntersectedSpans(4, 21, spanBorders)).toEqual([{from: 5, to: 6}, {from: 8, to: 10}, {from: 20, to: 21}]);
-    expect(getNonIntersectedSpans(2, 20, spanBorders)).toEqual([{from: 2, to: 3}, {from: 5, to: 6}, {from: 8, to: 10}]);
-    expect(getNonIntersectedSpans(2, 21, spanBorders)).toEqual([{from: 2, to: 3}, {from: 5, to: 6}, {from: 8, to: 10}, {from: 20, to: 21}]);
-    expect(getNonIntersectedSpans(3, 11, spanBorders)).toEqual([{from: 5, to: 6}, {from: 8, to: 10}]);
-    expect(getNonIntersectedSpans(3, 20, spanBorders)).toEqual([{from: 5, to: 6}, {from: 8, to: 10}]);
-    expect(getNonIntersectedSpans(3, 20, spanBorders)).toEqual([{from: 5, to: 6}, {from: 8, to: 10}]);
-    expect(getNonIntersectedSpans(4, 7, [3, 5, 6, 8])).toEqual([{from: 5, to: 6}]);
+
+function getCuts(from: number, to: number, xs: [number, number][]): [number, number][] {
```
Comment (rozetko) [inline server/spec/utils/spans.jest.ts]: what `xs` stands for?

File: server/src/controllers/analytics_controller.ts (hunk 2 above)
Comment (jonyrock) [reply server/src/controllers/analytics_controller.ts]: @rozetko as soon as we can legally cast `{from, to}` to a Span, why not?

**Pull request merged**